### ISSUE TITLE
Changing Where Images Are Stored

### ISSUE THREAD

**Issue by StopTrackingUs** [YunoHost-Apps/pixelfed_ynh#65]
This is a request to change where photos are stored to the /home folder.  Peertube does this.  I think it works great for those who set up pixelfed after they have already deployed yunohost for other purposes.  The issue they could run into is the /var folder not having enough space, because by default debian makes home the biggest directory.  For our server there is 3.6TB in /home and 95GB in /var lol.  I didn't realize this until I started getting disk space notifications.

**Comment by StopTrackingUs** [comment]
The exact place of storage used by peertube is /home/yunohost.app/peertube/storage. Which is really handy with the issue above.

**Comment by lapineige** [comment]
Does symlinking the folder work ?

Maybe one could add on option during install (and then in a config file) that specify the storage folder.
That would allow people to change it if needed, to their preferable location.

(For instance in my case the biggest partition is /media, as I have an external storage)

**Comment by yalh76** [comment]
@StopTrackingUs 
I agree with you and I have seen that when I repackaged peertube but the difference is that peertube configuration file provide a way to personalise storage:

* https://github.com/YunoHost-Apps/peertube_ynh/blob/6deb157214c0bee7423d5e4bbb4c49798dd41b27/conf/production.yaml#L57-L69

* https://github.com/Chocobozzz/PeerTube/blob/838f4a798395006233528969474d94e1c5f52293/config/default.yaml#L62-L74

and pixelfed doesn't provide that kind of functionality... Maybe you can open an issue to ask them to be able to choose storage location in the .env file. And when available we can implemnt it in pixelfed_ynh

I actually don't know where photos are currently stored :///

@lapineige 

too much options kill options ;)

btw when we will move storage to /home/yunohost.app/pixelfed/storage, you would be able to move to the storage to your /media folder and make a symbolic link  ;)

In conclusion, if you agree, as long as the option to choose the storage is not implemented in pixelfed I close this issue.

**Comment by lapineige** [comment]
> too much options kill options ;)

Well, that's basically the same as configuring the .env (except it's easier for the non technical users ^^)


But I agree :)

**Comment by yalh76** [comment]
> Well, that's basically the same as configuring the .env (except it's easier for the non technical users ^^)

But non-technical users even don't what is a linux folder or which path to provide :p
So this option would be more for mid-advanced users or lazy power root admins :ppp

**Comment by StopTrackingUs** [comment]
Well the issue for this admittidely lazy admin is that I have a pretty big peertube installation installed on the same server that has been running for almost a year.  In order to resize the disk I have to take it off line.  So requesting someone else do the work was my work around for not having down time lol.

But I can help with the set up.  I would guess most people that are using yunohost as their first debian based system wouldn't realize in the installation process that they need a big var or to split between var and home.

**Comment by StopTrackingUs** [comment]
I talked to dansup and he said they are working on s3 support.  Which I would expect would require implementations to be able to customize storage locations.  I'll close this one.  Let me know if I can pitch in on packaging or anything.

**Comment by yalh76** [comment]
> 
> 
> Well the issue for this admittedly lazy admin is that I have a pretty big peertube installation installed on the same server that has been running for almost a year. In order to resize the disk I have to take it off line. So requesting someone else do the work was my work around for not having down time lol.
> 
> But I can help with the set up. I would guess most people that are using yunohost as their first debian based system wouldn't realize in the installation process that they need a big var or to split between var and home.

Sorry you misunderstood me, the lazy power root admin was for @lapineige. for the option to choose the location during install. And for me being lazy for an IT is a quality :)

And I totally agree to move the storage to /home/yunohost.app/pixelfed/storage

**Comment by yalh76** [comment]
> 
> 
> I talked to dansup and he said they are working on s3 support. Which I would expect would require implementations to be able to customize storage locations. I'll close this one. Let me know if I can pitch in on packaging or anything.

Good news :) I don't took the time to speak with upstream application maintainers
